bin/s3stalker_daemon.py

```diff
@@ -0,0 +1,176 @@
+            else:
+                seconds_back = self._get_seconds_back_to_search(last_fetch_time)
+                config['timedelta'] = {'seconds': seconds_back}
+
+            messages = create_messages_for_recent_files(self.bucket, config)
+
+            last_fetch_time = get_last_fetch()
+            logger.info("Last fetch time...: %s", str(last_fetch_time))
+
+            for message in messages:
+                logger.info("Publishing %s", str(message))
+                self.publisher.send(str(message))
+
+            waiting_time = timedelta(**self.time_back)
+            wait_seconds = waiting_time.total_seconds()
+
+            # Wait for some time...
+            logger.debug("Waiting %d seconds", wait_seconds)
+            time.sleep(max(wait_seconds, 0))
+
+    def _get_seconds_back_to_search(self, last_fetch_time):
+        """Update the time to look back considering also the modification time of the last file."""
+        dtime_back = timedelta(**self.time_back)
+        seconds_back = dtime_back.total_seconds()
+
+        if last_fetch_time is None:
+            return seconds_back
+
+        start_time = datetime.utcnow()
+        start_time = start_time.replace(tzinfo=timezone.utc)
+        seconds_to_last_file = (start_time - last_fetch_time).total_seconds()
+        return max(seconds_back, seconds_to_last_file)
+
+    def close(self):
+        """Shutdown the S3Stalker runner."""
+        logger.info('Terminating the S3 Stalker daemon/runner.')
+        self.loop = False
+        if self.publisher:
+            try:
+                self.publisher.stop()
+            except Exception:
+                logger.exception("Couldn't stop publisher.")
+
+
+def arg_parse():
+    """Handle input arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("bucket", help="The bucket to retrieve from.")
+    parser.add_argument("config", help="Config file to be used")
+    parser.add_argument("-s", "--startup_timedelta_seconds",
+                        type=int,
+                        help="The time window in seconds back in time on start up (default=3600)",
+                        default=3600)
+    parser.add_argument("-l", "--log",
+                        help="Log configuration file",
+                        default=None)
+
+    return parser.parse_args()
+
+
+def main():
+    """Stalk an s3 bucket."""
+    args = arg_parse()
+
+    bucket = args.bucket
+    config = read_yaml(args.config)
+    startup_timedelta_seconds = args.startup_timedelta_seconds
+
+    if args.log is not None:
+        with open(args.log) as fd:
+            log_dict = yaml.safe_load(fd.read())
+            logging.config.dictConfig(log_dict)
+
+    logger.info("Try start the s3-stalker runner:")
+    try:
+        s3runner = S3StalkerRunner(bucket, config, startup_timedelta_seconds)
+
+    except Exception as err:
+        logger.error('The S3 Stalker Runner crashed: %s', str(err))
+        sys.exit(1)
```

Comment:
I think is a bit excessive when checking that only the initialization is failing and the stalker hasn't even yet started. Maybe merge this and the next try/except block?

Reply:
Right, thanks, done